docs/DeveloperGuide.md

```diff
@@ -1,34 +1,283 @@
+ 
+ ![End command sequence diagram](./pictures/endActivity.PNG)
+
+ **Ending a continued activity:**
+ 
+When the user wants to end the continued activity, an *EndCommand* object is created in *Parser.* The method *executeCommand()* of the *ContinueCommand* object is then called and it in turn executes the *saveActivity()* method of the *ActivityList* class. The continued activity is then saved by executing the following:
+
+ 1. Gets the current time and saves it to a public static variable of *Parser* object
+2.  Calculates the elapsed time using *between()* method of *Duration* class
+3.  Adds the elapsed time with the previous duration of the activity to get the *newDuration* using *plus()* method of Duration class
+4.  Calls the *updateDuration()* method, which updates the *duration* attribute of the continued activity in the *activityList* as well as the data.csv file
+
+#### 3.5.2 Design Considerations
+
+**Execution:**
+ - Continue by activity name (current implementation)
+ **Cons:** Activity names have to be unique.
+ **Pros:** More versatile, resistant to changes in the activity list
+ - Continue by activity index
+ **Cons:** need to add an additional index field to the Activity class, 
+ index is not fixed, changes when an activity is deleted
+ **Pros:** Can reuse activity names.
+ 
+Although the current implementation of the continue feature disallows users to have multiple activities with the same name, we felt that the versatility of this choice outweighed the cons. Firstly because if the activityList got too big, it would be hard for the user to get the index of the task he/she wanted to continue. Also, the index would constantly be changing when changes are made to the list.
+
+#### 3.5.3 Additional Features
+As users can only have activities with unique names, when a user wants to start an activity which already exists in the activityList, he/she will be given the option to continue the stated activity.
+![decision flowchart](./pictures/continue_flowchart.PNG)
+
+### 3.6 Find Feature
+This command accepts a keyword and searches the activity list for activities with names that contain the keyword.
+
+#### 3.6.1 Current Implementation
+* This feature is called by the user when the “find” command is entered into the command line. The string following the command is the keyword to match activity names to.
+* The Parser will create a FindCommand object.
+* The FindCommand will invoke its own executeCommand() method.
+    * The Parser's lastShownList will be cleared.
+    * Then it will loop through activityList to find activities with names that contain the keyword.
+    * If one is found, it will be added to lastShownList.
+    * printResults() of the Ui will be called:
+        * If lastShownList is not empty, it will print the matching activities.
+        * Else, it will respond to the user that there are no tasks which match the given keyword.
+
+
+![find seq diagram](https://imgur.com/Icg5rdB.png)
```

Comment:
This is a good use of sequence diagram. Perhaps you can reference from this when making future changes.